### PR TITLE
ZCL_ABAPGIT_SERIALIZE / DETERMINE_MAX_PROCESSES refactoring

### PR DESCRIPTION
**Pull request by christianguenter2**
- simplify conditional logic
- add more unit tests
- mock exit

### CODE REVIEW

File: src/zif_abapgit_environment.intf.abap [hunk 1]
```diff
@@ -34,4 +34,10 @@ INTERFACE zif_abapgit_environment
   METHODS is_variant_maintenance
     RETURNING
       VALUE(rv_is_variant_maintenance) TYPE abap_bool.
+  METHODS init_parallel_processing
+    IMPORTING
+      iv_group                      TYPE rzlli_apcl
```
Comment (larshp) [inline src/zif_abapgit_environment.intf.abap:39]: so the INTF is ABAP cloud compatible, 

```suggestion
      iv_group                      TYPE clike
```

Comment (christianguenter2) [reply src/zif_abapgit_environment.intf.abap]: Thx. Fixed.
How can I detect these kind of problems in advance?

**Comment by mbtools** [comment]
the UTs fail because I'm using an exit:

image

```abap
  METHOD zif_abapgit_user_exit~change_max_parallel_processes.
    cv_max_processes = 4.
  ENDMETHOD.

  METHOD zif_abapgit_user_exit~change_rfc_server_group.
    cv_group = 'abapgit'.
  ENDMETHOD.
```

Maybe add a check...

```abap
  METHOD determine_max_processes_free.

    CHECK mo_cut->mv_group = 'parallel_generators'. "<<<

    given_parallel_proc_disabled( abap_false ).
    ...
```

**Comment by christianguenter2** [comment]
> the UTs fail because I'm using an exit:
> 
> image
> 
> ```abap
>   METHOD zif_abapgit_user_exit~change_max_parallel_processes.
>     cv_max_processes = 4.
>   ENDMETHOD.
> 
>   METHOD zif_abapgit_user_exit~change_rfc_server_group.
>     cv_group = 'abapgit'.
>   ENDMETHOD.
> ```
> 
> Maybe add a check...
> 
> ```abap
>   METHOD determine_max_processes_free.
> 
>     CHECK mo_cut->mv_group = 'parallel_generators'. "<<<
> 
>     given_parallel_proc_disabled( abap_false ).
>     ...
> ```

Good catch, thx.

Hm, wouldn't be mocking/doubling the exit the right thing to do?

**Comment by fabianlupa** [comment]
Yes I had the same experience in the original PR. I'd suggest disabling calls to exit implementations in test contexts (factory method could probably check for sy-repid or if a test class is accessible). Bonus would be mocking in tests.
I am more concerned of someone implementing an exit and someone else executing the "harmless" unit tests in the same system.

**Comment by mbtools** [comment]
mocking would work. it's just more complicated since it's not forseen yet.

**Comment by christianguenter2** [comment]
Ok, let's start with mocking the exit. Disabling calls to exit implementation in test context isn't as easy as it seems. E.g. `SY-CPROG` depends on various factors (SAPGUI/ADT/NW release...). But I agree, that's something we should think about.

**Review by mbtools — APPROVED** [review]
Works great. Thanks

**Pull request merged**